R/textanal.R

```diff
@@ -631,19 +685,62 @@ tidy.textmodel_lda_exploratory <- function(x, type = "doc_topics", num_top_words
     res <- terms_topics_df %>% dplyr::group_by(topic) %>% dplyr::slice_max(probability, n = num_top_words, with_ties = FALSE) %>% dplyr::ungroup()
   }
   else if (type == "doc_topics") {
-    res <- x$df
-    docs_topics_df <- as.data.frame(x$model$theta)
-    docs_topics_df <- docs_topics_df %>% dplyr::mutate(max_topic = summarize_row(across(starts_with("topic")), which.max), topic_max = summarize_row(across(starts_with("topic")), max))
-    res <- res %>% dplyr::bind_cols(docs_topics_df)
-  }
-  else if (type == "doc_topics_mds") {
-    res <- x$df[x$docs_sample_index,]
-    docs_topics_sampled <- x$model$theta[x$docs_sample_index,]
-    docs_topics_df <- as.data.frame(docs_topics_sampled)
-    docs_topics_df <- docs_topics_df %>% dplyr::mutate(max_topic = summarize_row(across(starts_with("topic")), which.max))
-    res <- res %>% dplyr::bind_cols(docs_topics_df)
-    docs_coordinates_df <- as.data.frame(x$docs_coordinates)
-    res <- res %>% dplyr::bind_cols(docs_coordinates_df)
+    res <- x$doc_df
+  }
+  else if (type == "doc_topics_tagged") {
+    words_to_tag_df <- x$doc_word_df %>% distinct(document, word, .keep_all = TRUE)
+    words_to_tag_df <- words_to_tag_df %>% dplyr::mutate(max_topic = summarize_row(across(starts_with("topic")), which.max.safe), topic_max = summarize_row(across(starts_with("topic")), max))
+    words_to_tag_df <- words_to_tag_df %>% dplyr::group_by(document) %>% dplyr::slice_max(topic_max, prop=0.3) %>% dplyr::ungroup() # Filter per document.
+    tag_df <- words_to_tag_df %>% dplyr::nest_by(document) %>% ungroup()
+    res <- x$doc_df %>% rename(text=!!x$text_col) %>% mutate(doc_id=row_number()) %>% left_join(tag_df, by=c("doc_id"="document"))
```

Comment:
Add namespace to distinct, mutate, ungroup, rename and select.

Reply:
Done.